src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,46 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<K, V> {
+    private K first;
+    private V second;
 
+    private Pair(K first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public K getFirst() {
+        return first;
+    }
+
+    public V getSecond() {
+        return second;
+    }
+
+    public static <K, V> Pair<K, V> of(K first, V second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        Pair pair = (Pair) other;
```

Comment:
when we know the types we work with in the method, we should use them
in these rare cases we may ignore IDEA warnings

this is our method implementation, and we can ensure ourselves that this cast is secure
```suggestion
        Pair<K, V> pair = (Pair<K, V>) other;
```

Reply:
done